cryostatio/cryostat-helm
# feat(deploy): minimal parameter for deploying Cryostat without grafana and datasource

Fixes #35 

- Added `minimal` parameter to allow deploying Cryostat without a Grafana + JFR Data Source instance.
- Regenerated README and value schema with `readme-generator` (v2.5.0).
- Updated `NOTES.txt` to only exclude Grafana in minimal mode.
- Updated tests to to exclude Grafana health check in minimal mode.
- Updated `Chart.yaml` appVersion to be `v2.4.0-dev` (matches that `"cryostatVersion":"v2.4.0-dev-32-gf9a7f00f"` returned by health endpoint).

## How to test

1. Install chart with minimal parameter

```bash
helm install cryostat-sample ./cryostat/ --set minimal=true
```

2. Check if the user note does not mention Grafana network configurations

```
NAME: cryostat-sample
LAST DEPLOYED: Mon May 29 03:47:33 2023
NAMESPACE: myns
STATUS: deployed
REVISION: 1
NOTES:
1. Tell Cryostat how to serve external traffic:
  \```
  export POD_NAME=$(kubectl get pods -n myns -l "app.kubernetes.io/name=cryostat,app.kubernetes.io/instance=cryostat-sample"  --sort-by=.metadata.creationTimestamp -o jsonpath="{.items[-1:].metadata.name}")
  export CONTAINER_PORT=$(kubectl get pod -n myns $POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
  kubectl -n myns set env deploy --containers=cryostat cryostat-sample CRYOSTAT_WEB_HOST=127.0.0.1 CRYOSTAT_EXT_WEB_PORT=8080
  \```

2. Forward local ports to the application's pod:
  \```
  export POD_NAME=$(kubectl get pods -n myns -l "app.kubernetes.io/name=cryostat,app.kubernetes.io/instance=cryostat-sample"  --sort-by=.metadata.creationTimestamp -o jsonpath="{.items[-1:].metadata.name}")
  kubectl -n myns wait --for=condition=available --timeout=60s deploy/cryostat-sample
  kubectl -n myns port-forward $POD_NAME 8080:$CONTAINER_PORT
  \```

3. Visit the Cryostat application at: 
  \```
  http://127.0.0.1:8080
  \```
```

3. Check if cryostat pod only has the Cryostat container

```bash
kubectl get pods -n myns
NAME                              READY   STATUS    RESTARTS   AGE
cryostat-sample-76759987b-fzt4q   1/1     Running   0          3m5s
```

4. Run tests

```
helm test cryostat-sample --logs # Should exit 0
```
